### PR TITLE
[SPARK-43441][CORE] `makeDotNode` should not fail when DeterministicLevel is absent

### PR DESCRIPTION
**Pull request by TQJADE**
### What changes were proposed in this pull request?

This PR aims to make `makeDotNode` method handle the missing `DeterministicLevel`.

### Why are the changes needed?

Some old Spark generated data do not have that field.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Pass the CIs with newly added test case.

### CODE REVIEW

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you, @TQJADE .
Merged to master/3.4.

**Comment by dongjoon-hyun** [comment]
Welcome to the Apache Spark community, @TQJADE .
I added you to the Apache Spark contributor group and assigned SPARK-43441 to you.

File: core/src/main/scala/org/apache/spark/ui/scope/RDDOperationGraph.scala [hunk 1]
```diff
@@ -258,6 +258,7 @@ private[spark] object RDDOperationGraph extends Logging {
       case DeterministicLevel.DETERMINATE => ""
       case DeterministicLevel.INDETERMINATE => " [Indeterminate]"
       case DeterministicLevel.UNORDERED => " [Unordered]"
+      case _ => ""
```
Comment (mridulm) [inline core/src/main/scala/org/apache/spark/ui/scope/RDDOperationGraph.scala:261]: nit: We should have added a comment on why this is required - someone looking at the current codebase will not be able to understand how this is possible :-)

Comment (dongjoon-hyun) [reply core/src/main/scala/org/apache/spark/ui/scope/RDDOperationGraph.scala:261]: Yes, you're right. Unfortunately, this happens during loading old data generated by old Spark.

Could you make a follow-up PR, @TQJADE ?

**Pull request closed without merging**